core/arch/opnd.h

```diff
@@ -225,6 +229,24 @@ enum {
     DR_REG_XMM13,
     DR_REG_XMM14,
     DR_REG_XMM15,
+    DR_REG_XMM16,
+    DR_REG_XMM17,
+    DR_REG_XMM18,
+    DR_REG_XMM19,
+    DR_REG_XMM20,
+    DR_REG_XMM21,
+    DR_REG_XMM22,
+    DR_REG_XMM23,
+    DR_REG_XMM24,
+    DR_REG_XMM25,
+    DR_REG_XMM26,
+    DR_REG_XMM27,
+    DR_REG_XMM28,
+    DR_REG_XMM29,
+    DR_REG_XMM30,
+    DR_REG_XMM31,
+    /* 32 enums are reserved for future Intel SIMD extensions. */
+    RESERVED_XMM = DR_REG_XMM31 + 32,
```

Comment:
I don't know, I still suspect something is going to make this enum unable to keep contiguous ranges on some future update, no matter what slots are kept open here.  I don't want to have to change the enum again: that would make us look like bad software designers who can't create an interface that doesn't break every year.

Updating everything, including drreg's GPR indices, to use a function interface would make us happier long-term I think.  But I won't block this.  Just note that on the record I prefer a different approach.

Reply:
I can live with that :) After inspecting it for a while I think it would get more messy without the contiguous ranges. The ranges is nice to have internally. A compiler has it, too. It would become strange to have these fractured register classes. I thought being able to pull in the OPSZ_NA to 0 is actually a nice bonus in terms of cleanup. If you're semi-ok, I am fine with it!

- 

- 

-

Reply:
Another advantage: Pulling in the start of OPSZ_NA cleans up the partial overlap and makes more room for OPSZ variations. Right now, there are only a couple dozen left before 'byte' overflows.

Reply:
Please add a comment like `XXX i#3528: Switch from guaranteed-contiguous exposed enum ranges, which are not possible to maintain long-term, to function interfaces.` somewhere.